src/plugins/query_enhancements/public/search/sql_async_search_interceptor.ts

```diff
@@ -71,33 +78,43 @@ export class SQLAsyncSearchInterceptor extends SearchInterceptor {
       ...dataFrame.meta,
       queryConfig: {
         ...dataFrame.meta.queryConfig,
-        ...(this.connectionsService.getSelectedConnection() &&
-          this.connectionsService.getSelectedConnection()?.dataSource && {
-            dataSourceId: this.connectionsService.getSelectedConnection()?.dataSource.id,
-          }),
+        ...(this.connectionsService.getSelectedConnection() && {
+          dataSourceId: this.connectionsService.getSelectedConnection()?.id,
```

Comment:
I don't know how expensive `this.connectionsService.getSelectedConnection()` is but it would be a lot easier to understand if we just did:

```
const selectedConnection = this.connectionsService.getSelectedConnection();
if (selectedConnection) dataFrame.meta.queryConfig.dataSourceId = selectedConnection.id;

Reply:
In the updates from #7368 we don't depend on `connectionsService` at all